packages/dynamic-links/e2e/dynamicLinks.e2e.js

```diff
@@ -58,171 +60,334 @@ const getShortLink = async function (url, type) {
+          e.code.should.containEql('not-found');
+          e.message.should.containEql('Dynamic link not found');
+          return Promise.resolve();
+        }
+      });
+
+      it('throws on static links', async function () {
+        const { getDynamicLinks, resolveLink } = dynamicLinksModular;
+        try {
+          await resolveLink(getDynamicLinks(), TEST_LINK2_TARGET);
+          return Promise.reject(new Error('Did not throw Error.'));
+        } catch (e) {
+          e.message.should.containEql('Dynamic link not found');
+          return Promise.resolve();
+        }
+      });
+
+      it('throws on invalid links', async function () {
+        const { getDynamicLinks, resolveLink } = dynamicLinksModular;
+        try {
+          await resolveLink(getDynamicLinks(), null);
+          return Promise.reject(new Error('Did not throw Error.'));
+        } catch (e) {
+          e.message.should.containEql('Invalid link parameter');
+          return Promise.resolve();
+        }
+      });
+
+      // // The API is documented as being capable of suffering a processing failure, and we
+      // // handle it, but I don't know how to trigger it to validate
+      // it('throws on link processing error', async () => {
+      //   const { getDynamicLinks, resolveLink } = dynamicLinksModular;
+      //   try {
+      //     await resolveLink(getDynamicLinks(), SOME UNKNOWN INPUT TO CAUSE PROCESSING ERROR);
+      //     return Promise.reject(new Error('Did not throw Error.'));
+      //   } catch (e) {
+      //     e.code.should.containEql('resolve-link-error');
+      //     return Promise.resolve();
+      //   }
+      // });
+    });
 
+    describe('getInitialLink()', function () {
+      it('should return the dynamic link instance that launched the app', async function () {
+        const { ShortLinkType } = dynamicLinksModular;
+        const shortLink = await getShortLink(TEST_LINK3_TARGET, ShortLinkType.SHORT);
         if (device.getPlatform() === 'android') {
-          await device.launchApp({ url: shortLink });
+          await device.launchApp({ newInstance: true, url: shortLink });
         } else {
-          await device.openURL({ url: TEST_LINK1 });
+          // #2660 - iOS getInitialLink fails, while Linking.getInitialLink() will return the link for resolution !?
+          // #2631 - on iOS getInitialLink will return same link: open app w/link (correct), background, open again (old link)
+          await device.openURL({ url: TEST_LINK3 });
+          // TODO: ios is not able to open short links on simulator?
+          // await device.openURL({ url: shortLink });
         }
-        await Utils.spyToBeCalledOnceAsync(spy);
 
-        const link = spy.getCall(0).args[0];
+        // Since the app is relaunched and resumed after the initial import,
+        // the references to local variables are invalid. Therefore, we do another import here.
+        const { getDynamicLinks, getInitialLink } = dynamicLinksModular;
```

Comment:
import on this line: https://github.com/invertase/react-native-firebase/pull/7296/files?diff=split#diff-b885fc4a786a779f37b59dd276f88b9625d573aee097b53e5d8b4f47789a306aR370